### PR TITLE
Shortcut sol

### PR DESCRIPTION
**Pull request by HarryWangATX**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-899.mdx [hunk 1]
```diff
@@ -111,4 +111,115 @@ int main() {
...
+
+		long res = 0L;
+		for (int i = 1; i < farms.length; i++) {
+			if(dist[i] <= T) continue;
+			res = Math.max(res, (dist[i] - T) * farms[i]);
+		}
+
+		io.println(res);
+		io.close();
+	}
```
Comment (SansPapyrus683) [inline solutions/usaco-899.mdx:194]:
```suggestion
	}

```

File: solutions/usaco-899.mdx [hunk 2]
```diff
@@ -111,4 +111,115 @@ int main() {
...
+	public static long dfs(int t) {
+		for (Edge e : adj[t]) {
+			farms[t] += dfs(e.t);
+		}
+
+		return farms[t];
+	}
+
+	private static class Edge implements Comparable<Edge> {
```
Comment (SansPapyrus683) [inline solutions/usaco-899.mdx:203]: codesnip this

File: solutions/usaco-899.mdx [hunk 3]
```diff
@@ -111,4 +111,115 @@ int main() {
...
+		}
+
+		io.println(res);
+		io.close();
+	}
+	public static long dfs(int t) {
+		for (Edge e : adj[t]) {
+			farms[t] += dfs(e.t);
+		}
+
```
Comment (SansPapyrus683) [inline solutions/usaco-899.mdx:199]:
```suggestion
```

File: solutions/usaco-899.mdx [hunk 4]
```diff
@@ -111,4 +111,115 @@ int main() {
...
+				if(dist[e.t] <= nw) {
+					continue;
+				}
+
+				pq.add(new Edge(c.t, e.t, nw));
+			}
+		}
+		fin.remove(0);
+
+		for (int i = 0; i < N; i++) {
```
Comment (SansPapyrus683) [inline solutions/usaco-899.mdx:174]: extreme lack of comments

File: solutions/usaco-899.mdx [hunk 5]
```diff
@@ -111,4 +111,115 @@ int main() {
...
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class Shortcut {
+	static int N, M, T;
+	static long[] farms, dist;
```
Comment (SansPapyrus683) [inline solutions/usaco-899.mdx:123]: N, M, T, and `dist` don't need to be global

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-899.mdx [hunk 6]
```diff
@@ -111,4 +111,119 @@ int main() {
...
+		farms = new long[N];
+		adj = new List[N];
+
+		for (int i = 0; i < N; i++) {
+			farms[i] = io.nextInt();
+			dist[i] = Integer.MAX_VALUE;
+			adj[i] = new ArrayList<>();
+		}
+
+		// I/O Work
```
Comment (SansPapyrus683) [inline solutions/usaco-899.mdx:140]:
```suggestion
```
comments need not state the obvious

File: solutions/usaco-899.mdx [hunk 7]
```diff
@@ -111,4 +111,119 @@ int main() {
...
+		for (int i = 1; i < farms.length; i++) {
+			if(dist[i] <= T) continue;
+			res = Math.max(res, (dist[i] - T) * farms[i]);
+		}
+
+		io.println(res);
+		io.close();
+	}
+
+	public static long dfs(int t) {
```
Comment (SansPapyrus683) [inline solutions/usaco-899.mdx:200]: what does this dfs do?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

**Review by nwatx — APPROVED** [review]

**Pull request merged**